### PR TITLE
[AIRFLOW-5315] TaskInstance now only overwrites executor_config when explicitly told…

### PR DESCRIPTION
**Pull request by dimberman**
… to do so

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5315
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

Fixes a bug where a user is unable to update executor_configs for TaskInstances

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -454,7 +457,9 @@ def refresh_from_db(self, session=None, lock_for_update=False):
             self.max_tries = ti.max_tries
             self.hostname = ti.hostname
             self.pid = ti.pid
-            self.executor_config = ti.executor_config
+            # if update_executor_config:
```
Comment (gonzalodiaz) [inline airflow/models/taskinstance.py]: This commented line seems to be forgotten here :D

Comment (dimberman) [reply airflow/models/taskinstance.py]: good catch :P

**Review by kaxil — APPROVED** [review]
LGTM.

**Comment by kaxil** [comment]
I won't mind if we remove `executor_config` altogether from the TaskInstance table, so that it always uses the latest value. 


Currently, it is only used in k8s executor AFAIK. Might need a note in UPDATING.md

**Comment by dimberman** [comment]
@kaxil where would you want to move it?

**Comment by kaxil** [comment]
I mean I am not able to find a use-case where we want the `executor_config` value to be refreshed from DB, so I do not mind if we simply remove it from DB and by default always take the latest value. 

However, I am fine with the current change as well :)

**Pull request merged**